ygen/protogen.go

```diff
@@ -387,6 +403,71 @@ func genProto3Msg(msg *yangDirectory, msgs map[string]*yangDirectory, state *gen
 	return append(msgDefs, msgDef), errs
 }
 
+// writeProtoEnums takes a map of enumerations, described as yangEnum structs, and returns
+// the mapped protobuf enum definition that is required. It skips any identified enumerated
+// type that is a simple enumerated leaf, as these are output as embedded enumerations within
+// each message. It returns a slice of srings containing the generated code, and the slice
+// of errors experienced.
+func writeProtoEnums(enums map[string]*yangEnum) ([]string, []error) {
+	var errs []error
+	var genEnums []string
+	for _, enum := range enums {
+		if enum.entry.Type.Kind == yang.Yenum && enum.entry.Type.Name == "enumeration" {
+			// Skip simple enumerations.
+			continue
+		}
+
+		p := &protoEnum{Name: enum.name}
+		switch {
+		case enum.entry.Type.IdentityBase != nil:
```

Comment:
Define a helper and remove first sentence below?

Reply:
Done.